Add deprecation notice

martin308 opened the pull request:
Adds a deprecation notice and a link to the ruby beeline that people should use instead.

Remove all other information as it is not useful anymore and distracts from the deprecation notice

martin308 committed:
Add deprecation notice

martin308 committed:
Remove other information

This should make the deprecation notice stand out more

eanakashima reviewed (commented):
This gem is still our best answer for folks on rails 3, right?

In that case, 👍 for the big deprecation notice, but I'd love to keep at least the build status and the "Requires Ruby 2.2 or greater and Rails 3 or greater," line, so we can point people here if they need to use this. Maybe good to also have the deprecation notice specifically mention Rails 4 & 5 ?

martin308 commented:
Do we know if we have anyone using rails 3 with this gem? It's (rails 3) long past receiving any [support](https://guides.rubyonrails.org/maintenance_policy.html). If we do then sure we can add that back in along with that information. 

The reason I took it out was because the deprecation notice didn't really stand out very well with all of the extra content but we could play with the formatting a bit to make it stand out if we need to

eanakashima reviewed (approved):
Yeah, it's a good question. I'm used to people staying on old versions of rails years past the official support window from past jobs, but maybe that's changing. I don't know if we have any [current customers](https://ui-dogfood.honeycomb.io/prod/datasets/shepherd/result/eeqJ47ExjPh) on Rails 3. According to Honeycomb ([1](https://ui.honeycomb.io/ruby-together/datasets/rubygems.org/result/EhAnCPKdzm4), [2](https://ui.honeycomb.io/ruby-together/datasets/rubygems.org/result/uGuNjE16YCk)) about 4% of rails gem downloads are version 3.x, but they probably skew toward the kinds of companies that aren't ready to adopt Honeycomb.

You could always stick the "deprecation notice" bit in an H1, but then again we could also just take the rest of the content out and wait and see who complains — I have a mild preference for the former but think either's fine.

martin308 committed:
Keep more information in the readme

Keep some of the more useful information but move the deprecation notice
up higher to make it more visible
